pkg/proc/gdbserial/gdbserver.go

```diff
@@ -1426,7 +1435,11 @@ func (p *gdbProcess) updateThreadList(tu *threadUpdater) error {
 	}
 
 	for _, th := range p.threads {
-		if p.threadStopInfo {
+		shouldQueryStopInfo := p.threadStopInfo
+		if jstopInfo != nil {
+			_, shouldQueryStopInfo = jstopInfo[th.ID]
+		}
+		if shouldQueryStopInfo {
```

Comment:
I don't get this change. If we have a jsonstop record for the thread we do a second query to get a stock packet for the thread? We do this regardless of whether we know if the backend supports the qThreadStopInfo packet?

Reply:
Yeah, as of now I'm only limiting the number of stopInfo packets we send to the threads listed in jstopinfo. That way we don't issue stop info packets for every thread, every time, only the ones we know are in some interesting state.

There is a further optimization which I've noted in the TODO comment where we use only the jstopinfo packet when present and avoid sending any stopInfo packets at all. I've held off on doing that in this PR as the change will be more involved and I didn't want to add too much noise to this patch, so I'm going to send that in as a follow up.

Reply:
However I did fix the issue with not taking support of `qThreadStopInfo` into account.